wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -231,11 +243,23 @@ public Response create(
           @DefaultValue("false")
           Boolean startAfterCreate,
       @ApiParam("Namespace where workspace should be created") @QueryParam("namespace")
-          String namespace)
+          String namespace,
+      @Context HttpHeaders headers)
       throws ConflictException, BadRequestException, ForbiddenException, NotFoundException,
           ServerException {
     requiredNotNull(devfile, "Devfile");
 
+    DevfileImpl devfileModel;
+    try {
+      if (headers.getRequestHeader(HttpHeaders.CONTENT_TYPE).contains(APPLICATION_JSON)) {
```

Comment:
why don't we create two separate methods one for yaml other for json?

Reply:
They would differ only in the way the string is parsed. I didn't want to duplicate the many annotations on the rest method (and add the burden of having to remember to keep them in sync) just for the sake of slightly cleaner code.